src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/NotificationRequestMapper.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.mapper;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.helper.ContestedCourtHelper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.notification.NotificationRequest;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONSENTED_SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_SOLICITOR_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.DIVORCE_CASE_NUMBER;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.GENERAL_EMAIL_BODY;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_REFERENCE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_REFERENCE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.CommonFunction.isConsentedApplication;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.CommonFunction.isContestedApplication;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class NotificationRequestMapper {
```

Comment:
I would think twice making it a service. Is it going to be used by more than one service? Does it need dependencies injected? Both answers seem to be "no" and it would point into a direction of a helper class rather.

Reply:
> Is it going to be used by more than one service?
Possibly, if we ever decide break out our notification service into say a consented and contested notification service.

> Does it need dependencies injected?
Not currently, possibly in the future. I don't think needing dependencies should be our definition for a service.

> into a direction of a helper class rather
The issue with helper classes (as mentioned in the best practices) is that every unit test that calls the helper class (in this call all of them) would need to support the test data to also test these methods. If the logic inside these methods was to change, any unit tests in caller methods would also start failing. In this scenario we have unit test coverage of these methods in one place. The caller methods mock the response and verify they have been invoked. That's why I've suggested moving away from helper classes altogether and use these base services to improve our testing